sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -847,16 +993,24 @@ def __init__(self, descriptor, data_channel_factory, counter_factory,
                 runner=beam_fn_api_pb2.StateKey.Runner(key=token)),
             element_coder_impl))
 
-  _known_urns = {}
+  _known_urns = {}  # type: Dict[str, Tuple[ConstructorFn, Union[Type[message.Message], Type[bytes], None]]]
 
   @classmethod
-  def register_urn(cls, urn, parameter_type):
+  def register_urn(cls,
+                   urn,  # type: str
+                   parameter_type  # type: Optional[Type[T]]
+                  ):
+    # type: (...) -> Callable[[Callable[[BeamTransformFactory, str, beam_runner_api_pb2.PTransform, T, Dict[str, List[operations.Operation]]], operations.Operation]], Callable[[BeamTransformFactory, str, beam_runner_api_pb2.PTransform, T, Dict[str, List[operations.Operation]]], operations.Operation]]
```

Comment:
Not until we get to python3 annotations :(

Reply:
There are ways to create module-level aliases to break down and hide complexity, but since this uses a bunch of TypeVars to create relationships between the args and return value I had to do it all at once.